Controllers/userController.js

```diff
@@ -0,0 +1,248 @@
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt');
+const User = require('../models/User');
+const mailgun = require('mailgun-js');
+const { sendSuccess, sendError } = require('../utils/responseHandler');
+const uniqueToken = require('../utils/tokenGenerator');
+
+const regController = async (req, res) => {
+  // VALIDATE USER BEFORE SAVE
+  const { error, value } = await User.regValidations(req.body);
+  if (error) {
+    return sendError(res, error.details[0].message);
+  }
+
+  //Destructure filtered value...
+  const { username, email } = value;
+
+  //CHECK IF EMAIL EXIST IN THE DATABASE
+  const emailExist = await User.findOne({ email });
+  if (emailExist) {
+    const message = 'Email already exist, try another one';
+    return sendError(res, [], message);
+  }
+
+  // CREATE AND ASSIGN A TOKEN
+  const token = uniqueToken();
+
+  // CREATE EMAIL VERICATION MESSAGE
+  const link = `<div>
+      <p>Hey ${username}
+
+      <p>Verification Token</p>
+
+      <h3>${token}</h3>
+
+      <p>All The Best</p>
+    </div>
+    `;
+
+  //Save new user to database
+  const user = new User({ ...value, token });
+  const newUser = await user.save();
+  if (!newUser) {
+    const message = 'Error! Try again';
+    return sendError(res, [], message);
+  }
+
+  //EMAIL CONFIGURATION
+  const mg = await mailgun({
+    apiKey: process.env.MAILGUN_API_KEY,
+    domain: process.env.MAILGUN_DOMAIN,
+  });
+  const data = {
+    from: '"youngprinnce 👻" <' + process.env.EMAIL_ADDRESS + '>', // sender address
+    to: `${email}`, // list of receivers
+    subject: 'Verification Token ✔', // Subject line
+    text: 'hello', // plain text body
+    html: link,
+  };
+  await mg.messages().send(data, (err, body) => {
+    if (err) {
+      const message = 'Error! Try again';
+      return sendError(res, err, message);
+    } else {
+      const message = `Verification token has been sent to ${email}`;
+      return sendSuccess(res, [], message);
+    }
+  });
+};
```

Comment:
Create a service/function for sending email, so that you can always call the function anytime you want to send email
sample.
```
function sendEmail(receiver, message)
{
  //your mailgun cose shoudbe here
}
```

Try to make your code reusable.
DRY(don't repeat yourself) principle

Reply:
fixed